system/jlib/jutil.cpp

```diff
@@ -3282,7 +3282,8 @@ bool checkCreateDaemon(unsigned argc, const char * * argv)
 #ifndef _CONTAINERIZED
     for (unsigned i=0;i<(unsigned)argc;i++) {
         if (streq(argv[i],"--daemon") || streq(argv[i],"-d")) {
-            if (daemon(1,0) || write_pidfile(argv[++i])) {
```

Comment:
Why the extra check (seems a reasonable check), but if you are improving the error checking it should also be checking that i+1 < argc

Reply:
I'm just checking that the argument after --daemon isn't another flag.  You're absolutely right, I should add a check for i + 1 < argc.

Reply:
With the extra check you are preventing the filename from starting with a -, but you should also probably have an equivalent check in daserver.cpp.  It seems an unnecessary extra test.